[BUG]: CI Check for color usage in vectors will never pass until minSdk is updated

### Describe the bug

File Content Validation CI check that prohibits usages of colors other than component_colors in drawable files fails correctly on CI when violated, but the pattern that is being enforced cannot be supported by our current codebase, since we have not yet removed support for kitkat.

The CI error is:
```
app/src/main/res/drawable/rounded_primary_button_grey_shadow_color.xml:5: Only colors from component_colors.xml may be used in drawables except vector assets.
app/src/main/res/drawable/survey_nps_radio_selected_color.xml:4: Only colors from component_colors.xml may be used in drawables except vector assets.
app/src/main/res/drawable/survey_nps_radio_selected_color.xml:8: Only colors from component_colors.xml may be used in drawables except vector assets.
app/src/main/res/drawable/survey_nps_radio_unselected_color.xml:4: Only colors from component_colors.xml may be used in drawables except vector assets.
app/src/main/res/drawable/survey_nps_radio_unselected_color.xml:8: Only colors from component_colors.xml may be used in drawables except vector assets.
app/src/main/res/values/color_palette.xml:256: All color declarations in component_color.xml and color_palette.xml should end with _color suffix following snake_case naming convention.
```

Upon applying the recommended fix, the project build fails with error:
```
Cause 1: org.gradle.workers.internal.DefaultWorkerExecutor$WorkExecutionException: A failure occurred while executing com.android.build.gradle.internal.tasks.Workers$ActionFacade
	...
Caused by: java.lang.RuntimeException: Error while processing /Users/adhiambooyier/opensource/oppia-android/app/src/main/res/drawable/survey_nps_radio_selected_color.xml : Can't process attribute android:fillColor="@color/component_color_survey_primary_button_background_color": references to other resources are not supported by build-time PNG generation.
File was preprocessed as vector drawable support was added in Android 5.0 (API level 21)
See http://developer.android.com/tools/help/vector-asset-studio.html for details.
.android.ide.common.resources.MergedResourceWriter$FileGenerationWorkAction.run(MergedResourceWriter.java:412)
	... 34 more
```

### Steps To Reproduce

In any vector file in res/drawable, replace a `<path>` tag `fillColor` attribute color hex (e.g. #ffffff) with a color resource (e.g. @color/component_color_shared_screen_secondary_background_color) and then rebuild the app.

### Expected Behavior

I expect the project to build without errors.

### Screenshots/Videos

![Screenshot 2023-06-27 at 05 24 29](https://github.com/oppia/oppia-android/assets/59600948/5af99b96-9ff2-401e-943e-fc4235e4ebc8)



### What device/emulator are you using?

_No response_

### Which Android version is your device/emulator running?

_No response_

### Which version of the Oppia Android app are you using?

_No response_

### Additional Context

This  check was introduced in #4960

--- a/scripts/assets/file_content_validation_checks.textproto
+++ b/scripts/assets/file_content_validation_checks.textproto
@@ -354,7 +354,7 @@ file_content_checks {
 }
 file_content_checks {
   file_path_regex: "app/src/main/res/drawable.*?/.+?\\.xml"
-  prohibited_content_regex: "@color/(?!component_color_).+|\"#\\p{XDigit}+\""
+  prohibited_content_regex: "@color/(?!component_color_).+"
   failure_message: "Only colors from component_colors.xml may be used in drawables except vector assets."
   exempted_file_patterns: "app/src/main/res/drawable.*?/(ic_|lesson_thumbnail_graphic_).+?\\.xml"
   exempted_file_patterns: "app/src/main/res/drawable/full_oppia_logo.xml"

--- a/scripts/assets/file_content_validation_checks.textproto
+++ b/scripts/assets/file_content_validation_checks.textproto
@@ -353,17 +353,6 @@ file_content_checks {
   failure_message: "Only colors from component_colors.xml may be used in layouts."
 }
 file_content_checks {
-  file_path_regex: "app/src/main/res/drawable.*?/.+?\\.xml"
-  prohibited_content_regex: "@color/(?!component_color_).+"
-  failure_message: "Only colors from component_colors.xml may be used in drawables except vector assets."
-  exempted_file_patterns: "app/src/main/res/drawable.*?/(ic_|lesson_thumbnail_graphic_).+?\\.xml"
-  exempted_file_patterns: "app/src/main/res/drawable/full_oppia_logo.xml"
-  exempted_file_patterns: "app/src/main/res/drawable/rounded_white_background_with_shadow.xml"
-  exempted_file_patterns: "app/src/main/res/drawable/profile_image_shadow.xml"
-  exempted_file_patterns: "app/src/main/res/drawable/selected_region_background.xml"
-  exempted_file_patterns: "app/src/main/res/drawable/splash_page.xml"
-}
-file_content_checks {
   file_path_regex: "app/src/main/java/org/oppia/android/app.?/.+(ActivityPresenter|FragmentPresenter|ViewPresenter|Activity|Fragment|View)\\.kt"
   prohibited_content_regex: "R.color.(?!component_color_).+"
   failure_message: "Only colors from component_colors.xml may be used in Kotlin Files (Activities, Fragments, Views and Presenters)."

--- a/scripts/src/javatests/org/oppia/android/scripts/regex/RegexPatternValidationCheckTest.kt
+++ b/scripts/src/javatests/org/oppia/android/scripts/regex/RegexPatternValidationCheckTest.kt
@@ -2345,6 +2345,7 @@ class RegexPatternValidationCheckTest {
       )
   }
 
+  // TODO(#5075): Add hex color into prohibitedContent for drawables
   @Test
   fun testFileContent_xmlDrawables_includesNonColorComponentReferences_fileContentIsNotCorrect() {
     val prohibitedContent =
@@ -2352,7 +2353,6 @@ class RegexPatternValidationCheckTest {
         android:color="@color/component_color_shared_primary_text_color"
         android:color="@color/color_defs_shared_primary_text_color"
         android:color="@color/color_palette_primary_text_color"
-        android:color="#003933"
       """.trimIndent()
     tempFolder.newFolder("testfiles", "app", "src", "main", "res", "drawable")
     val stringFilePath = "app/src/main/res/drawable/test_layout.xml"
@@ -2369,7 +2369,6 @@ class RegexPatternValidationCheckTest {
         """
         $stringFilePath:2: $doesNotReferenceColorFromComponentColorInDrawables
         $stringFilePath:3: $doesNotReferenceColorFromComponentColorInDrawables
-        $stringFilePath:4: $doesNotReferenceColorFromComponentColorInDrawables
         $wikiReferenceNote
         """.trimIndent()
       )

--- a/scripts/src/javatests/org/oppia/android/scripts/regex/RegexPatternValidationCheckTest.kt
+++ b/scripts/src/javatests/org/oppia/android/scripts/regex/RegexPatternValidationCheckTest.kt
@@ -173,8 +173,6 @@ class RegexPatternValidationCheckTest {
     "Only colors from color_defs.xml may be used in color_palette.xml."
   private val doesNotReferenceColorFromComponentColorInLayouts =
     "Only colors from component_colors.xml may be used in layouts."
-  private val doesNotReferenceColorFromComponentColorInDrawables =
-    "Only colors from component_colors.xml may be used in drawables except vector assets."
   private val doesNotReferenceColorFromComponentColorInKotlinFiles =
     "Only colors from component_colors.xml may be used in Kotlin Files (Activities, Fragments, " +
       "Views and Presenters)."
@@ -2345,34 +2343,7 @@ class RegexPatternValidationCheckTest {
       )
   }
 
-  // TODO(#5075): Add hex color into prohibitedContent for drawables
-  @Test
-  fun testFileContent_xmlDrawables_includesNonColorComponentReferences_fileContentIsNotCorrect() {
-    val prohibitedContent =
-      """
-        android:color="@color/component_color_shared_primary_text_color"
-        android:color="@color/color_defs_shared_primary_text_color"
-        android:color="@color/color_palette_primary_text_color"
-      """.trimIndent()
-    tempFolder.newFolder("testfiles", "app", "src", "main", "res", "drawable")
-    val stringFilePath = "app/src/main/res/drawable/test_layout.xml"
-    tempFolder.newFile("testfiles/$stringFilePath").writeText(prohibitedContent)
-
-    val exception = assertThrows(Exception::class) {
-      runScript()
-    }
-
-    // Verify that all patterns are properly detected & prohibited.
-    assertThat(exception).hasMessageThat().contains(REGEX_CHECK_FAILED_OUTPUT_INDICATOR)
-    assertThat(outContent.toString().trim())
-      .isEqualTo(
-        """
-        $stringFilePath:2: $doesNotReferenceColorFromComponentColorInDrawables
-        $stringFilePath:3: $doesNotReferenceColorFromComponentColorInDrawables
-        $wikiReferenceNote
-        """.trimIndent()
-      )
-  }
+  // TODO(#5075): Add test for drawables file that checks color uses only component colors
 
   @Test
   fun testFileContent_kotlinFiles_includesNonColorComponentReferences_fileContentIsNotCorrect() {

